docs/grpc_client.md

```diff
@@ -0,0 +1,53 @@
+# gRPC client example
+
+We use a gRPC API for interacting with LNDK. To pay an offer, you'll need to
+connect to the LNDK server with a gRPC client, which you can do in any language. [LINK] 
+
+Please note that since LNDK connects to LND, you'll need to pass in your LND
+credentials (tls certificate, macaroon, and grpc server address). The client
+must pass in this data via gRPC metadata.
+
+An example client in Rust:
+
+```
+use offers::offers_client::OffersClient;
+use offers::PayOfferRequest;
+use tonic::metadata::MetadataValue;
+use tonic::transport::Channel;
+use tonic::Request;
+
+pub mod offers {
+    tonic::include_proto!("offers");
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let channel = Channel::from_static("http://[::1]:7000").connect().await?;
+
+    let tls_cert_path: MetadataValue<_> =
+        "<TLS_CERT_PATH>".parse()?;
+    let macaroon_path: MetadataValue<_> =
+        "<MACAROON_PATH>".parse()?;
+    let address: MetadataValue<_> =
+        "<LND_SERVER_ADDRESS>".parse()?;
+
+    let mut client = OffersClient::with_interceptor(channel, move |mut req: Request<()>| {
+        req.metadata_mut().insert("tls_cert_path", tls_cert_path.clone());
+        req.metadata_mut().insert("macaroon_path", macaroon_path.clone());
+        req.metadata_mut().insert("address", address.clone());
+        Ok(req)
+    });
```

Comment:
This should work in the same way as the LND gRPC client which supports passing the macaroon as a hex encoded string, and the cert as a PEM encoded string.

e.g.

cert string
```
-----BEGIN CERTIFICATE-----
MIICRzCCAeygAwIBAgIQBdrxyaexLDcWO+l32rUTuTAKBggqhkjOPQQDAjA4MR8wHQYDVQQKExZsbmQgYXV0b2dlbmVyYXRlZCBjZXJ0MRUwEwYDVQQDEwxkNDA0YzI5ZTI5OTMwHhcNMjQwNTA1MjIwMTE4WhcNMjUwNjMwMjIwMTE4WjA4MR8wHQYDVQQKExZsbmQgYXV0b2dlbmVyYXRlZCBjZXJ0MRUwEwYDVQQDEwxkNDA0YzI5ZTI5OTMwWTATBgcqhkjOPQIBBggqhkjOPQMBBwNCAAQ27rYuASfE8M/5nE6JLULK8voDiHXxuAdOMKl/e4wfEKlDA8e+K1H+J4IJH45OXaj/6BdAtsyhX3R+d6lP3iAAo4HXMIHUMA4GA1UdDwEB/wQEAwICpDATBgNVHSUEDDAKBggrBgEFBQcDATAPBgNVHRMBAf8EBTADAQH/MB0GA1UdDgQWBBQ3gSzsrSmXLX9Oq19OHl8GWzWJHDB9BgNVHREEdjB0ggxkNDA0YzI5ZTI5OTOCCWxvY2FsaG9zdIIOb2x5bXB1cy1sbmR1czGCBHVuaXiCCnVuaXhwYWNrZXSCB2J1ZmNvbm6HBH8AAAGHEAAAAAAAAAAAAAAAAAAAAAGHBKwcAQGHEP6AAAAAAAAAAEKs//4cAQEwCgYIKoZIzj0EAwIDSQAwRgIhAI2ZS5xoafjuWpSvj9Ts/6twV1Dg3uyeO7v7/0dyLumWAiEAsxDH59RPAk0EJCcV4szhnfibjP8Bwzi3dfDrn+VgfEU=
-----END CERTIFICATE-----
```

macaroon string:
```
0201036C6E6402F801030A1034F41C28A3B5190702FEA607DD4045AE1201301A160A0761646472657373120472656164120577726974651A130A04696E666F120472656164120577726974651A170A08696E766F69636573120472656164120577726974651A210A086D616361726F6F6E120867656E6572617465120472656164120577726974651A160A076D657373616765120472656164120577726974651A170A086F6666636861696E120472656164120577726974651A160A076F6E636861696E120472656164120577726974651A140A057065657273120472656164120577726974651A180A067369676E6572120867656E6572617465120472656164000006205CF3E77A97764FB2A68967832608591BE375B36F51A6269AB425AA767BC22B55
```

This would neable b etter interoperability between LND and LNDK.

Reply:
Hmm yeah so we're restricted by tonic_lnd -- for now, it only allows passing in the macaroon/tls file paths: https://github.com/fedimint/tonic_lnd

But if doing it this way is essential for Strike, LMK. I think I've seen other people request the same feature, so we could probably explore adding the option upstream and, after that, in LNDK

Reply:
Yes, we'd need this change. Otherwise, we'd end up having to patch our setup to take security obtained credentials and save them on the disk to be able to use this, which would require a bunch of nasty workarounds. And others attempting to use this will likely run into the same issue and have to make similar workarounds in their own setup. Better that we address this once, here at the source.

Reply:
Thinking on this some more, we're likely going to be connecting from non-rust (C#) client, so this is perhaps less important for our use case.

Reply:
Sounds good, I actually just remembered that someone else implemented this already here: https://github.com/Kixunil/tonic_lnd/pull/30 Though we'll need to move it over to the fedimint fork that's still maintained.

IOW, I'll try to get this working in LNDK soon :)